### PR TITLE
AttributeError raised when using response_model in FastAPI route decorator #21744

### PR DESCRIPTION
**Pull request by hannahcoles**
## Why are these changes needed?
This adds an argument input check that prevents FastAPI type errors. 

## Related issue number
Closes #21744

## Important Note
I wrote the following regression test in order to see if my code worked and if the issue was solved. After running it, I got the error message "fastapi.exceptions.FastAPIError: Invalid args for response field! Hint: check that [4.81, 4.81, 4.81, 4.81] is a valid pydantic field type." It appears that fastapi already accounts for invalid arguments for response_field. Therefore, I'm not sure if you still want to accept the arg check, as it isn't necessarily needed. As my regression test has failed due to the FastAPIError, I did not include it in the pull request. I can update my pull request with the test if needed.
`

    app = FastAPI()
    class A:
        @app.get("/{i}")
        def b(self, i: int):
            pass
    
    bad_data = [4.81]*4

    @app.post('/fakepage/', response_model=bad_data)
    def z(self, i: int):
        pass

    make_fastapi_class_based_view(app, A)`

## Checks
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [x] This PR is not tested :(

### CODE REVIEW

**Comment by hannahcoles** [comment]
@simon-mo this is my pull request, let me know what you think.

**Comment by simon-mo** [comment]
Looks great! Would you mind adding a test case in https://github.com/ray-project/ray/blob/master/python/ray/serve/tests/test_fastapi.py?

**Comment by hannahcoles** [comment]
@simon-mo just added the test case!

File: python/ray/serve/tests/test_fastapi.py [hunk 1]
```diff
@@ -636,6 +636,30 @@ def decr2(self):
...
+    class Nested(BaseModel):
+        val: int
+
+    @app.post("/fakepage/", response_model=Nested)
+    def z(self, i: int):
+        pass
+
+    try:
+        make_fastapi_class_based_view(app, A)
+    except Exception:
```
Comment (simon-mo) [inline python/ray/serve/tests/test_fastapi.py:659]: No need for exception and fail here. If the previous call throw the test fails.

**Review by simon-mo — APPROVED** [review]

**Comment by edoakes** [comment]
Looks like there's a test failure here:
https://buildkite.com/ray-project/ray-builders-pr/builds/30150#bbb543d4-9b86-46f5-a1a2-1efc5ea0c265/242-4315

**Comment by pcmoritz** [comment]
Should we push this over the finish line @simon-mo ?

**Pull request closed without merging**